health-services/geopode-adapter/src/main/java/digit/util/ArcgisUtil.java

```diff
@@ -0,0 +1,147 @@
+
+@Component
+@Slf4j
+public class ArcgisUtil {
+    private Configuration config;
+    private RestTemplate restTemplate;
+    private MdmsV2Util mdmsV2Util;
+    @Autowired
+    private ObjectMapper mapper;
+
+    public ArcgisUtil(Configuration config, RestTemplate restTemplate, MdmsV2Util mdmsV2Util) {
+        this.config = config;
+        this.restTemplate = restTemplate;
+        this.mdmsV2Util=mdmsV2Util;
+    }
+
+    public BoundaryResponse createRoot(GeopodeBoundaryRequest request) {
+        MdmsResponseV2 mdmsResponse = fetchMdmsData(request);
+        Optional<String> countryName = extractCountryNameFromMdms(mdmsResponse, request.getGeopodeBoundary().getISOCode());
+
+        if (countryName.isEmpty()) {
+            throw new CustomException("COUNTRY_NAME_NOT_FOUND", "No country found for ISO code: " + request.getGeopodeBoundary().getISOCode());
+        }
+
+        fetchArcGisData(countryName.get()); // For now we are not using the response (e.g., geometry/rings)
+
+        BoundaryRequest boundaryRequest = buildBoundaryRequest(countryName, config.getTenantId(), request);
+        return sendBoundaryRequest(boundaryRequest);
+    }
+
+
+    private MdmsCriteriaReqV2 buildMdmsV2RequestForRoot(GeopodeBoundaryRequest request) {
+        return MdmsCriteriaReqV2.builder()
+                .requestInfo(request.getRequestInfo())
+                .mdmsCriteriaV2(MdmsCriteriaV2.builder()
+                        .tenantId(config.getTenantId())
+                        .schemaCode(config.getSchemaCode())
+                        .offset(Integer.parseInt(config.getDefaultOffset()))
+                        .limit(Integer.parseInt(config.getDefaultLimit()))
+                        .build())
+                .build();
+    }
+
+    private MdmsResponseV2 fetchMdmsData(GeopodeBoundaryRequest request) {
+        MdmsCriteriaReqV2 mdmsCriteriaReqV2 = buildMdmsV2RequestForRoot(request);
+        String url = config.getMdmsHost() + config.getMdmsV2EndPoint();
+
+        try {
+            return restTemplate.postForObject(url, mdmsCriteriaReqV2, MdmsResponseV2.class);
+        } catch (Exception e) {
+            log.error(NO_MDMS_DATA_FOUND_FOR_GIVEN_TENANT_ISO_CODE, e);
+            throw new CustomException("MDMS_FETCH_FAILED", "Failed to fetch MDMS data");
+        }
+    }
+
+
+    private BoundaryRequest buildBoundaryRequest(Optional<String> countryName, String tenantId, GeopodeBoundaryRequest request){
+        return BoundaryRequest.builder()
+                .requestInfo(request.getRequestInfo())
+                .boundary(List.of(
+                        Boundary.builder()
+                                .code(countryName.orElse(null))
+                                .tenantId(config.getTenantId())
+                                .build()
+                ))
+                .build();
+    }
+
+    private Optional<String> extractCountryNameFromMdms(MdmsResponseV2 mdmsResponse, String targetIsoCode) {
```

Comment:
why is it returning Optional, it should not retun optional. throw exception if there is no ISOCode